src/_common/content/content-viewer/content-viewer.ts

```diff
@@ -79,6 +94,43 @@ export default class AppContentViewer extends Vue implements ContentOwner {
 	setContent(content: ContentDocument) {
 		this.data = content;
 		this.hydrator = new ContentHydrator(content.hydration);
+		const processedItems: any[] = [];
+
+		this.data.content.forEach(item => {
+			if (item.type === 'mediaItem' && !item.attrs.href) {
+				processedItems.push(
+					new AppContentMediaItem({
+						propsData: {
+							mediaItemId: item.attrs.id,
+							mediaItemWidth: item.attrs.width,
+							mediaItemHeight: item.attrs.height,
+							caption: item.attrs.caption,
+							align: item.attrs.align,
+							href: item.attrs.href,
+							isEditing: false,
+							owner: this.owner,
+						},
+					}).mediaItem
+				);
+			}
+		});
+
+		if (this.lightbox) {
+			// Assign to the temp array if the user is currently looking at lightbox items.
+			this.mediaItemsTemp = processedItems;
+		} else {
+			this.mediaItems = processedItems;
+			this.mediaItemsTemp = null;
+		}
+	}
+
+	setActiveItem(item: any) {
+		let index = item;
+		if (typeof item === 'object') {
```

Comment:
Wouldn't the item need to match the signature of a `MediaItem`? Ie we should be able to check `instanceof MediaItem`, right?